frontends/smv_node.h

```diff
@@ -10,30 +11,1276 @@
+  element_node * trans_li;
+  element_node * invar_li;
+  element_node * invarspec_li;
+
+ public:
+  module_node(std::string name) { module_name = name; }
+  module_node(std::string name, std::vector<string> id_list)
+  {
+    module_name = name;
+    par_li = id_list;
+  }
+  module_node(std::string name,
+              std::vector<string> id_list,
+              std::unordered_map<SMVnode::NodeMtype, element_node *> decl_map)
+  {
+    module_name = name;
+    par_li = id_list;
+    var_li = new element_node();
+    ivar_li = new element_node();
+    frozenvar_li = new element_node();
+    define_li = new element_node();
+    assign_li = new element_node();
+    init_li = new element_node();
+    trans_li = new element_node();
+    invar_li = new element_node();
+    invarspec_li = new element_node();
+    for (std::unordered_map<SMVnode::NodeMtype, element_node *>::iterator it =
+             decl_map.begin();
+         it != decl_map.end();
+         ++it) {
+      if (it->first == SMVnode::DEFINE) define_li = it->second;
+      if (it->first == SMVnode::ASSIGN) assign_li = it->second;
+      if (it->first == SMVnode::IVAR) ivar_li = it->second;
+      if (it->first == SMVnode::VAR) var_li = it->second;
+      if (it->first == SMVnode::FROZENVAR) frozenvar_li = it->second;
+      if (it->first == SMVnode::INIT) init_li = it->second;
+      if (it->first == SMVnode::TRANS) trans_li = it->second;
+      if (it->first == SMVnode::INVAR) invar_li = it->second;
+      if (it->first == SMVnode::INVARSPEC) invarspec_li = it->second;
+    }
+  }
+
+  module_node(std::string name,
+              std::unordered_map<SMVnode::NodeMtype, element_node *> decl_map)
+  {
+    module_name = name;
+    var_li = new element_node();
+    ivar_li = new element_node();
+    frozenvar_li = new element_node();
+    define_li = new element_node();
+    assign_li = new element_node();
+    init_li = new element_node();
+    trans_li = new element_node();
+    invar_li = new element_node();
+    invarspec_li = new element_node();
+    for (std::unordered_map<SMVnode::NodeMtype, element_node *>::iterator it =
+             decl_map.begin();
+         it != decl_map.end();
+         ++it) {
+      if (it->first == SMVnode::DEFINE) define_li = it->second;
+      if (it->first == SMVnode::ASSIGN) assign_li = it->second;
+      if (it->first == SMVnode::IVAR) ivar_li = it->second;
+      if (it->first == SMVnode::VAR) var_li = it->second;
+      if (it->first == SMVnode::FROZENVAR) frozenvar_li = it->second;
+      if (it->first == SMVnode::INIT) init_li = it->second;
+      if (it->first == SMVnode::TRANS) trans_li = it->second;
+      if (it->first == SMVnode::INVAR) invar_li = it->second;
+      if (it->first == SMVnode::INVARSPEC) invarspec_li = it->second;
+    }
+  }
+  unordered_map<string, SMVnode *> get_namelist() { return new_par; }
+  std::string get_par() { return par_name; }
+  void pre(std::unordered_map<std::string, module_node *> module_list,
```

Comment:
Could you add a comment explaining the difference between `pre` and `preprocess`?